src/providers/node/hash.ts

```diff
@@ -0,0 +1,105 @@
+import * as crypto from 'crypto';
+import Promise from 'dojo-core/Promise';
+import { Binary, Codec, Data, Hasher, HashFunction } from '../../crypto';
+
+/**
+ * Algorithms available through this provider. This
+ */
+export const ALGORITHMS = {
```

Comment:
Is this exported with the intent of allowing users to create their own provider that builds off of this as a base?  (This would likely be something we'd want to explain how to do eventually.)